### PR TITLE
[DCA11Y-1989] fix loosely defined majors

### PR DESCRIPTION
**Pull request by gt-lassian**
**Summary**

When a node version is defined as `v20` or `v22` or just `20`, the plugin's `install-node-and-npm` step will fail with error:
```
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.15.1-atlassian-3:install-node-and-npm (step-1-set-up-environment) on project react: Range [0, -1) out of bounds for length 3 
```

**Tests and Documentation**

I'm reusing the previously commented out method `findMatchingReleasedVersion` with a fallback to the `"v" + version` (as before. Also, tests are added, mocking the positive and negative paths from `findMatchingReleasedVersion`.

### CODE REVIEW

File: frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/NodeVersionHelper.java [hunk 1]
```diff
@@ -228,7 +228,11 @@ public static String getDownloadableVersion(String version) {
 
         version = version.replaceFirst("v", ""); // we're about to add it back
 
-//        return findMatchingReleasedVersion(version)
+        Optional<String> matchingVersion = findMatchingReleasedVersion(version);
+        if (matchingVersion.isPresent()) {
```
Comment (rafalsatl) [inline frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/NodeVersionHelper.java:232]: when will this be false?

**Comment by gt-lassian** [comment]
OK! I think this is just a portion of the fix. The `v20` or just `20` should still be valid for various version managers. So the real failure is somewhere else.
